### ISSUE TITLE
opus audio file support

### ISSUE THREAD

**Issue by themarcnet** [qarmin/czkawka#943]
Hi,  First of all, great app - does what I want, when I want,  with no hassle....apart from one thing....

Would it be possible to add ".opus" as an audio format (extension is also .opus) that you scan the tags for.  Opus files use the same tagging as ogg files, which is no suprise since opus and ogg come from the same developers.    

I have many ogg, m4a, mp3, and opus files, and I would like to get rid of the duplicates (i.e: remove the ogg files that have the same tagging information as the newer opus files)

In fact, I can rename opus files to ogg, and your application can read the tags just fine!  Although it would be really nice for it to just work, without the need for this work-around.

More information about opus here -> https://github.com/xiph/opus

**Comment by mr-brune** [comment]
yes it would be usefull also add fingerprint duplicate finder for opus

**Comment by AKL55** [comment]
Would be nice to add as well.

**Comment by GitKopfkino** [comment]
This feature would be helpful!

**Comment by DocMAX** [comment]
+1

**Comment by Heavygrass** [comment]
+1

**Comment by Kilohertz77** [comment]
Replying to ask the exact same question as the OP did. 

Czkawka can already process Opus files when they have the .ogg extension, so please add .opus to the list of allowed extensions.

I now have a batch file set up that first renames all my .opus files to .ogg, then runs Czkawka, and then renames .ogg back to .opus. With a library of thousands of music files this takes too long and shouldn't be necessary.

**Comment by csolisr** [comment]
> Czkawka can already process Opus files when they have the .ogg extension, so please add .opus to the list of allowed extensions.

From what I see, adding `.opus` to the allowed extensions should be trivial, just a matter of appending ` "opus",` to this line: https://github.com/qarmin/czkawka/blob/f82bd5ddf4e1958651249a2b8e9db38df58eadb7/czkawka_core/src/common/consts.rs#L32

**Comment by csolisr** [comment]
(Also, for documentation purposes, to the following other lines): 

https://github.com/qarmin/czkawka/blob/f82bd5ddf4e1958651249a2b8e9db38df58eadb7/instructions/Instruction.md?plain=1#L538

https://github.com/qarmin/czkawka/blob/f82bd5ddf4e1958651249a2b8e9db38df58eadb7/czkawka_cli/src/commands.rs#L809

**Comment by Eoin-ONeill-Yokai** [comment]
@csolisr Have you tested this and, if so, care to make it a merge request?

Looking to fork the project myself soon to add this feature myself as it's really bothersome to not have opus support in my collection. So I'll be giving it a test on my end as well, but it looks like you're probably right on this one.

**Comment by csolisr** [comment]
I don't have the Rust compilation toolset installed in my computer yet, but I could try it sometime next week hopefully.

**Comment by quinn-p-mchugh** [comment]
Ahhh, I was confused when my library of 7,000 music files (of which I know there are many duplicates) only turned up with 10 or so. Glad to see others ran into this and that it seems to be a quick fix.

**Comment by Eoin-ONeill-Yokai** [comment]
@csolisr I've been running it on my end and it looks good to go, so I've made a pull request. I hope you don't mind.

If you'd prefer to upload it yourself, though, feel free and I'll give you priority for merge here as you technically did the legwork here. Just want you to know that your suggestion is working as expected. :)

**Comment by csolisr** [comment]
No prob for me as, again, I don't have the Rust build system installed yet - thanks!
